src/CloudNative.CloudEvents/Http/HttpUtilities.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright 2021 Cloud Native Foundation.
+// Licensed under the Apache 2.0 license.
+// See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CloudNative.CloudEvents.Http
+{
+    /// <summary>
+    /// Common functionality used by all Http code.
+    /// TODO: Can we make this internal again? It's useful in the ASP.NET Core package, but I don't really like having it public...
```

Comment:
Agree with questioning why this is public - especially the Encode/DecodeHeader methods which don't have anything to do with CloudEvents. I can see the usefulness of GetAttributeNameFromHeaderName though.

Reply:
It wasn't public until the ASP.NET Core package came along.

There's a lot of repetition in terms of how conversion to/from events is done, and I'm wondering whether there's a way of abstracting it out so that all our HTTP conversions (and possibly even MQTT, AMQP and Kafka) can just provide a generic adapter. I don't want to experiment with that until the core is more solid though :)

The header encoding is *a bit* CloudEvent specific, in that we need to follow the exact steps specified in the CloudEvents spec, which may not be the same set of requirements as other things. You'd expect it to be simple and handled by a single Framework method call, but no...